lib/src/folding/parsers/indent.dart

```diff
@@ -83,13 +82,19 @@ class IndentFoldableBlockParser extends AbstractFoldableBlockParser {
     _openBlocksLinesByIndent[indent] = lineIndex;
   }
 
-  void _closeBlocks(int i, int nextNonEmptyIndent) {
-    _openBlocksLinesByIndent.forEachInvertedWhile(
-      (indentsCount, startLine) {
-        _closeBlock(startLine, i);
-      },
-      executeWhile: (indentsCount, _) => indentsCount >= nextNonEmptyIndent,
-    );
+  void _closeBlocks(int index, int nextNonEmptyIndent) {
+    final keys = _openBlocksLinesByIndent.keys.toList();
+    for (int i = keys.length - 1; i >= 0; i--) {
```

Comment:
```suggestion
    for (final indent in _openBlocksLinesByIndent.keys.reversed) {
```
?

Make an extension getter `reversed` on Iterable. Make sure its `.toList()` is non-growable.
https://medium.com/flutter-senior/always-use-non-growable-arrays-if-possible-4864a022a54a